Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyEditorAPI_Internals.h

```diff
@@ -512,7 +512,15 @@ namespace AzToolsFramework
 
             const AZ::Uuid& desiredUUID = GetHandledType();
 
-            PropertyType* actualCast = static_cast<PropertyType*>(serializeContext->DownCast(tempValue, propertyType, desiredUUID));
+            PropertyType* actualCast;
+            if (serializeContext->CanDowncast(propertyType, desiredUUID))
+            {
+                actualCast = static_cast<PropertyType*>(serializeContext->DownCast(tempValue, propertyType, desiredUUID));
+            }
+            else
+            {
+                actualCast = serializeContext->Cast<PropertyType*>(tempValue, propertyType);
+            }
```

Comment:
I think you should initialize this in a lambda.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es28-use-lambdas-for-complex-initialization-especially-of-const-variables